torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -591,9 +575,84 @@ def _update_trainable(self) -> None:
         # Create the optim which will work on the param shard
         if not hasattr(self, "optim"):
             self._clear_cache()
-            self._default_device = list(self._per_device_params.keys())[0]
             self.optim = self._optim_constructor(self.partition_parameters()[self.rank], **self._optim_defaults)
             self._sync_param_groups(self.optim.param_groups, self.param_groups)
 
         if self.parameters_as_bucket_view:
             self._setup_flat_buffers()
+
+    def _verify_and_init_params(self, params: Any) -> None:
+        r"""
+        Verifies the type of ``params`` and initializes ``self._all_params``
+        if ``params`` is valid.
+
+        While :class:`optim.Optimizer <torch.optim.Optimizer>` allows
+        ``params`` to be an iterable of :class:`dict` s, currently
+        ``ZeroRedundancyOptimizer`` strictly requires ``params`` to be an
+        iterable of :class:`torch.Tensor` s.
+
+        Raises:
+            TypeError: ``params`` has an invalid type.
+            ValueError: ``params`` is empty.
+        """
+        if (isinstance(params, torch.Tensor)):
+            raise TypeError("params argument should be an iterable of "
+                            "Tensors, but got " + torch.typename(params))
```

Comment:
nit: this message and ones used below can use [f-strings](https://www.python.org/dev/peps/pep-0498/), e.g. (`f"Tensors, but got {torch.typename(params)}"`)

Reply:
Fixed.